tests/lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,321 @@
+//------------------------------------------------------------------------------
```

Comment:
Do you have any test cases for multiple classes in one file? Maybe some with correct usage and some with incorrect usage?

Also, have you tried testing this on a real codebase?

Reply:
I've tested this on our codebase of over 300 components and 200+ routes and it was able to identify and autofix all instances (109 usecases).

I'll add some test cases for nested and multiple classes to confirm proper identification and fixes for those.